packages/wpcom-checkout/src/partner-coupon.ts

```diff
@@ -15,9 +30,70 @@ export function getPartnerCoupon( {
 	coupon: string;
 	products?: ResponseCartProduct[];
 } ): boolean {
-	const productHasSublabel =
-		products && products.some( ( product: ResponseCartProduct ) => !! getSublabel( product ) );
+	const isAllowed =
+		products &&
+		products.some( ( product: ResponseCartProduct ) => !! isAllowedPartnerProduct( product ) );
 	const isPartnerCoupon = coupon.startsWith( 'IONOS_' );
 
-	return ( productHasSublabel && isPartnerCoupon ) || false;
+	return ( isAllowed && isPartnerCoupon ) || false;
+}
+
+function isAllowedPartnerProduct( product: ResponseCartProduct ): boolean {
```

Comment:
I'm not sure what this is meant to look for, so I just copied the logic previously found in `getSubLabel` and put it here. If we can figure out what we need to find, it would be nice to simplify this.